### DISCUSSION TITLE
Allow users to re-map or disable single character keyboard shortcuts

### DISCUSSION THREAD

**Discussion by aitchiss** [forem/forem#16224] [Feature Requests]
We currently have keyboard shortcuts on the home feed:

- `b`: saves/unsaves the focused article to your reading list
- `k`: changes focus to the previous article
- `j`: changes focus to the next article

As none of these shortcuts include modifier keys (like ctrl , shift or cmd), it's a [WCAG requirement](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html) that users are empowered to either turn off or re-map the shortcuts to other keys in their settings. The TL;DR is that any user who uses some sort of speech input technology will easily accidentally invoke these single character shortcuts.

Potential solutions would be:

- Change the shortcuts to include modifiers (taking care to avoid collisions with other browser shortcuts) e.g. ctrl+shift+k
- Allow users to switch the shortcuts off in settings
- Allow users to re-map the shortcuts in settings

In terms of the WCAG requirements, it's OK if the 'default' state is what we have now, as long as the mechanism exists to change it in their settings, but we could also consider having these shortcuts 'off' by default and letting users 'opt-in' in their settings.

**Comment by nickytonline** [comment]
One thing to be mindful of is how many people actually use these shortcuts. (I have no idea 🙃 ). If it's fairly low, I would be inclined to change the shortcut modifiers to avoid collisions. In the future, if there's a big demand for it, we could then revisit the possibility of folks customizing it via their settings.

The on/off in the settings could be a good choice as well.

**Comment by aitchiss** [comment]
I was going to create a separate issue for this, but I think the whole thing can/should be looked at holistically 😄 

While working on https://github.com/forem/forem/pull/16773 I noticed that we've implemented the `j` and `k` shortcuts by adding an additional, visually hidden, link to every story card in the feed. This is a whole lot of extra noise for screen reader users and keyboard navigators who _aren't_ using the `j`/`k` shortcuts. Given a story card already has _multiple_ links to the same location, there's no need for the additional noise. 

As we review these keyboard shortcuts, lets also keep this ☝️  in mind. We could refactor to remove these additional links, and instead target one of the visually available ones.
